TestSuite/contribution.py

```diff
@@ -0,0 +1,235 @@
+    ):
+        prefix = 'dashboard'
+        dashboard = self._create_json_based(name, prefix, content, dir_path=self._dashboards_path)
+        self.dashboards.append(dashboard)
+        return dashboard
+
+    def create_incident_field(
+            self,
+            name,
+            content: dict = None,
+            release_notes: bool = False
+    ):
+        prefix = 'incident-field'
+        incident_field = self._create_json_based(name, prefix, content, dir_path=self._incident_fields_path)
+        if release_notes:
+            release_notes = self._create_text_based(f'{incident_field}_CHANGELOG.md',
+                                                    dir_path=self._incidents_field_path)
+            self.incident_field.append(release_notes)
+        self.incident_field.append(incident_field)
+        return incident_field
+
+    def create_incident_type(
+            self,
+            name,
+            content: dict = None):
+        prefix = 'incident-type'
+        incident_type = self._create_json_based(name, prefix, content, dir_path=self._incident_types_path)
+        self.incident_types.append(incident_type)
+        return incident_type
+
+    def create_indicator_field(
+            self,
+            name,
+            content: dict = None
+    ):
+        prefix = 'incident-field'
+        indicator_field = self._create_json_based(name, prefix, content, dir_path=self._indicator_fields_path)
+        self.indicator_field.append(indicator_field)
+
+    def create_metadata_for_zip(self):
+        fake_metadata = {
+            "name": self.name,
+            "description": "",
+            "updated": "0001-01-01T00:00:00Z",
+            "created": "2020-06-30T10:35:51.24973515Z",
+            "support": "internalContribution",
+            "author": "Who Cares",
+            "authorImage": "",
+            "supportDetails": {
+                "url": "",
+                "email": "madeup@madeup.com"
+            }
+        }
+        self.metadata = JSONBased(self.target_dir, 'metadata', '')
+        self.metadata.write_json(fake_metadata)
+
+    def create_zip(self, zip_dst: Optional[Path] = None, del_src_files: bool = True):
+        self.create_classifier(name='fakeclassifier')
+        self.create_dashboard(name='fakedashboard')
+        self.create_incident_field(name='fakeincidentfield')
+        self.create_incident_type(name='fakeincidenttype')
+        self.create_indicator_field(name='fakeindicatorfield')
+        self.create_mapper(name='fakemapper')
+        self.create_integration()
+        self.create_script()
+        self.create_metadata_for_zip()
+        if zip_dst:
+            self.created_zip_filepath = shutil.make_archive(
+                zip_dst / self.name, 'zip', self.target_dir
+            )
+        else:
+            self.created_zip_filepath = shutil.make_archive(
+                self.target_dir.parent / self.name, 'zip', self.target_dir
+            )
+        if del_src_files:
```

Comment:
why is this needed?

Reply:
I don't want the source files polluting the temporary repo I'm the test is running in.